modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java

```diff
@@ -322,7 +322,7 @@ public void restoreMessagesFromDeadLetterQueue(@MBeanOperationParameter(name = "
             description = "The Dead Letter Queue Name for the selected tenant") String destinationQueueName) {
 
         if (null != andesMetadataIDs) {
-            List<Long> andesMessageIdList = Arrays.asList(ArrayUtils.toObject(andesMetadataIDs));
+            List<Long> andesMessageIdList = new ArrayList<>(Arrays.asList(ArrayUtils.toObject(andesMetadataIDs)));
```

Comment:
Suggestion may be we can do a addAll() -> removes a intermediate asList ?
If this doesn't hit performance-critical path I suggest we modify contract of caching layer.